pkg/api/scylla/v1alpha1/types_scylladbdatacenter.go

```diff
@@ -0,0 +1,528 @@
+// Copyright (c) 2024 ScyllaDB.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ScyllaDBDatacenterSpec defines the desired state of ScyllaDBDatacenter.
+type ScyllaDBDatacenterSpec struct {
+	// metadata controls shared metadata for all pods created based on this spec.
+	// +optional
+	Metadata ObjectTemplateMetadata `json:"metadata,omitempty"`
+
+	// clusterName specifies the name of the ScyllaDB cluster.
+	// When joining two DCs, their cluster name must match.
+	// This field is immutable.
+	ClusterName string `json:"clusterName"`
+
+	// datacenterName specifies the name of the ScyllaDB datacenter. Used as datacenter name in GossipingPropertyFileSnitch.
+	// If empty, it's taken from the 'scylladbdatacenter.metadata.name'.
+	// +optional
+	DatacenterName string `json:"datacenterName,omitempty"`
+
+	// scyllaDB holds a specification of ScyllaDB.
+	ScyllaDB ScyllaDB `json:"scyllaDB"`
+
+	// scyllaDBManagerAgent holds a specification of ScyllaDB Manager Agent.
+	// +optional
+	ScyllaDBManagerAgent *ScyllaDBManagerAgent `json:"scyllaDBManagerAgent,omitempty"`
+
+	// imagePullSecrets is an optional list of references to secrets in the same namespace
+	// used for pulling any images used by this spec.
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+
+	// dnsPolicy defines how a pod's DNS will be configured.
+	// +optional
+	DNSPolicy *corev1.DNSPolicy `json:"dnsPolicy,omitempty"`
+
+	// dnsDomains specifies a list of DNS domains this cluster is reachable by.
+	// These domains are used when setting up the infrastructure, like certificates.
+	// +optional
+	DNSDomains []string `json:"dnsDomains,omitempty"`
+
+	// forceRedeploymentReason specifies the latest redeployment reason.
+	// Can be used to force a rolling restart of all racks in this DC by providing a unique string.
+	// +optional
+	ForceRedeploymentReason string `json:"forceRedeploymentReason,omitempty"`
+
+	// exposeOptions specifies parameters related to exposing ScyllaDBDatacenter backends.
+	// +optional
+	ExposeOptions *ExposeOptions `json:"exposeOptions,omitempty"`
+
+	// rackTemplate provides a template for every rack.
+	// Every rack inherits properties specified in the template, unless it's overwritten on the rack level.
+	// +optional
+	RackTemplate *RackTemplate `json:"rackTemplate,omitempty"`
+
+	// racks specify the racks in the datacenter.
+	Racks []RackSpec `json:"racks"`
+
+	// disableAutomaticOrphanedNodeReplacement controls if automatic orphan node replacement should be disabled.
+	DisableAutomaticOrphanedNodeReplacement bool `json:"disableAutomaticOrphanedNodeReplacement,omitempty"`
```

Comment:
shouldn't it be marked as optional? From api convention: `Note that for backward compatibility, any field that has the omitempty struct tag will be considered to be optional, but this may change in the future and having the +optional comment tag is highly recommended.`

Reply:
changed to pointer and marked optional

Reply:
I don't see it as a pointer still - @zimnx are you sure you pushed the changes?